vault/kv_helpers.go

```diff
@@ -78,6 +78,27 @@ func kvReadRequest(client *api.Client, path string, params map[string]string) (*
 	return api.ParseSecret(resp.Body)
 }
 
+func kvListRequest(client *api.Client, path string) ([]interface{}, error) {
+	log.Printf("[DEBUG] Listing secrets at %s from Vault", path)
+	resp, err := client.Logical().List(path)
+	if err != nil {
+		return nil, fmt.Errorf("error listing from Vault at path %s, err=%s", path, err)
+	}
+
+	if resp == nil {
+		return nil, fmt.Errorf("no secrets found at %q", path)
```

Comment:
I believe that is possible that `Logical().List()` will return non `nil` response the the case of a 404 error, which is rather odd. It seems like it has a relationship to providing `Warnings` to the caller.

Ref: https://github.com/hashicorp/vault/blob/0dd4cda7c93d6b0fc7bf7d7b108c45a6fa7aaac6/api/logical.go#L127-L139

Returning an error if the `resp` is nil seems to make sense, but the client API doesn't do us any favors here.